src/core/trulens/core/tru.py

```diff
@@ -175,146 +134,27 @@ class Tru(python.SingletonPerName):
-        snowflake_connection_parameters: Optional[Dict[str, str]] = None,
-        app_name: Optional[str] = None,
-    ):
-        """
-        Args:
-            database_check_revision: Whether to check the database revision on
-                init. This prompt determine whether database migration is required.
-        """
-
-        if database_args is None:
-            database_args = {}
-
-        if snowflake_connection_parameters is not None:
-            if database is not None:
-                raise ValueError(
-                    "`database` must be `None` if `snowflake_connection_parameters` is set!"
-                )
-            if database_url is not None:
-                raise ValueError(
-                    "`database_url` must be `None` if `snowflake_connection_parameters` is set!"
-                )
-            if not app_name:
-                raise ValueError(
-                    "`app_name` must be set if `snowflake_connection_parameters` is set!"
-                )
-            schema_name = self._validate_and_compute_schema_name(app_name)
-            database_url = self._create_snowflake_database_url(
-                snowflake_connection_parameters, schema_name
-            )
-
-        database_args.update({
-            k: v
-            for k, v in {
-                "database_url": database_url,
-                "database_file": database_file,
-                "database_redact_keys": database_redact_keys,
-                "database_prefix": database_prefix,
-            }.items()
-            if v is not None
-        })
-
-        if python.safe_hasattr(self, "db"):
-            # Already initialized by SingletonByName mechanism. Give warning if
-            # any option was specified (not None) as it will be ignored.
-            for v in database_args.values():
-                if v is not None:
-                    logger.warning(
-                        "Tru was already initialized. Cannot change database configuration after initialization."
-                    )
-                    self.warning()
-                    break
-
-            return
-
-        if database is not None:
-            if not isinstance(database, DB):
-                raise ValueError(
-                    "`database` must be a `trulens.core.database.base.DB` instance."
-                )
-
-            self.db = database
-        else:
-            self.db = SQLAlchemyDB.from_tru_args(**database_args)
-
-        if database_check_revision:
-            try:
-                self.db.check_db_revision()
-            except DatabaseVersionException as e:
-                print(e)
-                self.db = OpaqueWrapper(obj=self.db, e=e)
-
-        # TODO: if snowflake_connection_parameters is not None:
-        #    # initialize stream for feedback eval table.
-        #    # initialize task for stream that will import trulens and try to run the Tru.
-
-    @staticmethod
-    def _validate_and_compute_schema_name(name):
```

Comment:
I don't see this method in snowflake connector. Is this missed?

Reply:
I can add the validation. As for computing the schema name, the snowflake connector takes a `schema_name` in place of the `app_name` so this method is not neccessary. Since `app_name` is defined in the AppDefinition and not Tru (as before), I felt having `app_name` as a Connector argument would conflate Apps and Connectors.